types/math/dec.go

```diff
@@ -33,6 +32,9 @@ var dec128Context = apd.Context{
 }
 
 func NewDecFromString(s string) (Dec, error) {
+	if s == "" {
+		s = "0"
+	}
```

Comment:
i found a few places where were we would explicitly check for the empty string case and then set it to "0" anyway, so it felt like a good way to reduce that.

are there any specific cases where we'd prefer to error here instead of default to 0?

Reply:
If we want to handle "", no need to parse anything just return a zero value. 

Seems like we'd check for "" elsewhere if we actually want a billable case so I think this default is okay